### PR TITLE
Added create date updating ability, refs #11820

### PR DESCRIPTION
**Pull request by mcantelon**
* Added ability to update actor history during CSV import
* Added ability to update event during CSV import:
  * Update if CSV date fields are empty
  * Update if CSV date fields are different than stored values

### CODE REVIEW

File: lib/task/import/csvImportBaseTask.class.php [hunk 1]
```diff
@@ -238,6 +238,37 @@ static function importEvents(&$import)
       // Add row culture to fetch place term in event creation/update
       $eventData['culture'] = $import->columnValue('culture');
 
+      // If in update mode and event's a creation event with an existing actor
+      // specified, see if the event seems like it should be updated
+      // rather than created.
+      if ($eventTypeId == QubitTerm::CREATION_ID && $import->matchAndUpdate
```
Comment (jraddaoui) [inline lib/task/import/csvImportBaseTask.class.php]: I think we should look for all event types, not just creation events. I'd remove that condition and add the type to the query directly.

File: lib/task/import/csvImportBaseTask.class.php [hunk 2]
```diff
@@ -238,6 +238,37 @@ static function importEvents(&$import)
...
+          && !empty($eventData['actorName'])
+          && null != $actor = QubitActor::getByAuthorizedFormOfName($eventData['actorName']))
+      {
+        // Check creation events to determine whether event should be updated rather than created
+        $criteria = new Criteria;
+        $criteria->add(QubitEvent::TYPE_ID, QubitTerm::CREATION_ID);
```
Comment (jraddaoui) [inline lib/task/import/csvImportBaseTask.class.php]: I'd compare with `$eventTypeId` to allow this behavior for all event types.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Hi @mcantelon,

This is looking great! I feel like we should do it for all event types, I couldn't see anything about the event type in the Redmine ticket.

Also, that ticket is a private one, we should create a public one and change it in the commit message that will be merged.

Thanks!

**Comment by jraddaoui** [comment]
I've created https://projects.artefactual.com/issues/12102

**Comment by mcantelon** [comment]
Good point re: event type and I'll change the issue reference in the commit to the public issue #... thanks!

**Comment by mcantelon** [comment]
I've now removed the logic that prevents it from working with non-creation events.

**Review by jraddaoui — APPROVED** [review]
Thanks @mcantelon!

**Pull request merged**